langchain4j-parent/pom.xml

```diff
@@ -436,6 +479,25 @@
                     <artifactId>maven-enforcer-plugin</artifactId>
                     <version>3.5.0</version>
                 </plugin>
+                <plugin>
+                    <groupId>com.diffplug.spotless</groupId>
+                    <artifactId>spotless-maven-plugin</artifactId>
+                    <version>2.43.0</version>
+                    <configuration>
+                        <!-- optional: limit format enforcement to just the files changed by this feature branch -->
+                        <ratchetFrom>origin/main</ratchetFrom>
+                        <java>
+                            <palantirJavaFormat>
+                                <version>2.50.0</version>
```

Comment:
AFAIU this is different from code style you've added in `.editorconfig`? Is there a way to use `.editorconfig` here as well?

Reply:
Not in palantirJavaFormat or googleJavaFormat
We can try https://github.com/ec4j/editorconfig-maven-plugin or update `.editorconfig` to match the palantir style, which is closer to Idea Style.

Reply:
- editorconfig-maven-plugin has very limited set of [supported .editorconfig properties](ttps://github.com/ec4j/editorconfig-maven-plugin?tab=readme-ov-file#how-it-works) and not suppporting custome `ij_*` settings,
- editorconfig-maven-plugin does not support ratchetFrom, so a Big-Bang reformat is required.

Not an option.

Reply:
So there is no point to have `.editorconfig` if we want to automate checks and reformatting? AFAIU there should be a single "source of truth"

Reply:
There is a point  to have `.editorconfig`: to help IDEs with formatting.
@langchain4j , I understand that palantirJavaFormat style might not be exactly what you want to see. I encourage you to tune the formatter and checker, so they meet your code style. No one could do it better than you 🤗